common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -644,6 +645,24 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
                 SpanExtension.current().setAttribute(
                         AttributeName.ccs_request_id.name(), ccsRequestId);
             }
+
+            final String ccsRequestSequence = response.getHeaderValue(XMS_CCS_REQUEST_SEQUENCE, 0);
+            if (null != ccsRequestSequence){
```

Comment:
Lines 649 - 665 look extremely similar to lines 631 - 644. Could some refactoring be done? For example, I think null checking tokenResponse and creating that mapWithAdditionalEntries could be done only once.

Reply:
+1 on this

Reply:
Refactored to include above suggestions